lib/routes/utils/oauth.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const ScopeSet = require('fxa-shared').oauth.scopes
+
+// right now we only care about notifications for the following scopes
+// if not a match, then we don't notify
+const NOTIFICATION_SCOPES = ScopeSet.fromArray(['https://identity.mozilla.com/apps/oldsync'])
+
+module.exports = {
+  newTokenNotification: async function newTokenNotification (db, oauthdb, mailer, devices, request, credentials, grant) {
+    const scopeSet = ScopeSet.fromString(grant.scope)
+
+    if (! scopeSet.intersects(NOTIFICATION_SCOPES)) {
+      // right now we only care about notifications for the `oldsync` scope
+      // if not a match, then we don't do any notifications
+      return
+    }
+
+    const tokenVerify = await oauthdb.checkAccessToken({
+      token: grant.access_token
+    })
```

Comment:
It seems like a bit of a code smell that you have to check the access_token here immediately after generating it; is this so that you can find out the `uid` in the `grant_type=authorization_code` case?  I wonder if we could arrange for the oauth-server to return that information directly from `/token` rather than having to verify it again here.

Reply:
Yeah that would be nice :)

Reply:
@rfk how about we leave this as a follow up for now? merge with the `checkAccessToken` then optimize by landing https://github.com/mozilla/fxa-auth-server/pull/3000 ?

Reply:
OK, works for me.